core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -460,6 +463,7 @@ protected Node(final Environment environment, Collection<Class<? extends Plugin>
                 b.bind(UsageService.class).toInstance(usageService);
                     b.bind(NamedWriteableRegistry.class).toInstance(namedWriteableRegistry);
                     b.bind(MetaDataUpgrader.class).toInstance(metaDataUpgrader);
+                    b.bind(TemplateUpgradeService.class).toInstance(templateUpgradeService);
```

Comment:
It seems this is only bound in Guice so that tests can get access to it, so they can check if upgrades are in progress. Is this something that should be exposed via an api? At least let's find another way so we can keep additional stuff out of guice.